service_thing_aggregate/src/test/scala/dao/ObjectSearchDaoSpec.scala

```diff
@@ -0,0 +1,246 @@
+    insert("C1", "21", "Fin øks")
+    insert("C2", "", "Sverd")
+
+    insert("C777", "35", "Øks")
+    insert("C.777", "34B", "Øks")
+    insert("C.777", "34", "Øks")
+    insert("C.777", "34A", "Øks")
+
+    insert("C555", "34B", "Øks")
+    insert("C555", "34A", "Øks")
+    insert("C555", "34C", "Øks")
+    insert("C555A", "B", "Øks")
+    insert("C555B", "A", "Øks")
+    insert("C555C", "C", "Øks")
+
+    insert("C888_B", "B", "Øks")
+    insert("C888_A", "A", "Øks")
+    insert("C888xC", "C", "Øks")
+
+    insert("C81%A", "A", "Bøtte")
+    insert("C81%XA", "B", "Bøtte")
+
+    insert("C81-A", "A", "Bøtte")
+    insert("C81-XA", "B", "Bøtte")
+
+    insert(s"C81${escapeChar}A", "A", "Bøtte")
+    insert(s"C81${escapeChar}XA", "B", "Bøtte")
+
+  }
+
+  "ObjectSearch" must {
+    "dummy test to insert test data" in {
+
+    }
+    "find an object which exists, via museumNo" in {
+      insertTestData(1)
+
+      val res = dao.search(1, "C1", "", "", 1, 100).futureValue
+      val seq = res.get
+      assert(seq.length >= 3)
+
+      val res2 = dao.search(1, "C2", "", "", 1, 100).futureValue
+      val seq2 = res2.get
+      assert(seq2.length > 0)
+
+    }
+
+    "offsets and sorting must work ok" in {
+
+      val res = dao.search(1, "C1", "", "", 1, 10).futureValue
+      val seq = res.get
+      seq.length mustBe 10
+
+      seq(0).subNo mustBe Some("1a")
+      seq(1).subNo mustBe Some("2a")
+
+      val res2 = dao.search(1, "C1", "", "", 2, 10).futureValue
+      val seq2 = res2.get
+      seq2.length mustBe 10
+
+      val res3 = dao.search(1, "C1", "", "", 3, 10).futureValue
+      val seq3 = res3.get
+
+      seq3.length mustBe 2
+
+      seq3(0).subNo mustBe Some("21")
+      seq3(1).subNo mustBe Some("22")
+    }
+  }
+
+  "museumNo with digits only" in {
+
+    val res = dao.search(1, "777", "", "", 1, 10).futureValue
+    val seq = res.get
+    seq.length mustBe 4
+
+    seq(0).subNo mustBe Some("34")
+    seq(1).subNo mustBe Some("34A")
+    seq(2).subNo mustBe Some("34B")
+    seq(3).subNo mustBe Some("35")
+  }
+
+  "museumNo with * should work" in {
+
+    val res = dao.search(1, "C555*", "", "", 1, 10).futureValue
+    val seq = res.get
+    seq.length mustBe 6
+
+    seq(0).subNo mustBe Some("34A")
+    seq(1).subNo mustBe Some("34B")
+    seq(2).subNo mustBe Some("34C")
+    seq(3).museumNo mustBe "C555A"
+    seq(4).museumNo mustBe "C555B"
+    seq(5).museumNo mustBe "C555C"
+
+  }
+
+  "simple SQL-injection attempt should return 0 rows" in {
+    val res = dao.search(1, "C.' or 1=1 --", "", "", 1, 10).futureValue
+    val seq = res.get
+    seq.length mustBe 0
+  }
+
+  "museumNo, subNo with * and term search should work" in {
+
+    val res = dao.search(1, "c555*", "3*", "øks", 1, 10).futureValue
+    val seq = res.get
+
+    seq.length mustBe 3
+    seq(0).subNo mustBe Some("34A")
+    seq(1).subNo mustBe Some("34B")
+    seq(2).subNo mustBe Some("34C")
```

Comment:
FYI, this can be written in a more concise way...although readability is probably not as good.

```scala
seq.map(_.subNo) must contain allOf(Some("34A"), Some("34B"), Some("34C"))
```